src/app/domain/case-viewer/components/summary-panel/summary-panel.component.ts

```diff
@@ -7,6 +8,26 @@ import { Component, Input } from '@angular/core';
 })
 export class SummaryPanelComponent {
     @Input() panelData;
+
+    @Input() caseId;
+
+    private getRecentEvents() {
```

Comment:
This seems a bit overly defensive and verbose. If we're processing events on change then we can simply splice a know array